app/models/manageiq/providers/openstack/network_manager/network_port.rb

```diff
@@ -1,2 +1,11 @@
 class ManageIQ::Providers::Openstack::NetworkManager::NetworkPort < ::NetworkPort
+  def disconnect_port
+    # Some ports link subnets to routers, so
+    # sever that association if the port is removed
+    cloud_subnets.each do |subnet|
+      subnet.network_router = nil
```

Comment:
So to correctly cover the latest spec. We would need info on network_port, to what router it relates to (ems_ref). Then here, we could check that if it's still the same router, we can nullify it. If it's a different router we should not touch it.

But it's kind of clunky. I think in the end, we should look into how to change the modeling. So that 1 model doesn't have to touch relations of other models. So e.g. doing the actual relation through NetworkPort and deleting the subnet.network_router foreign key.

Reply:
The subnet gets network_router set on it as part of parsing network ports. I take your comment about timing mean that the whole inventory is collected and parsed, and then subnets are saved before network ports, resulting in the property being cleared?

We can't compare whether the router is the same, because the user could delete and recreate the same router relationship. That would shrink the edge case though. I agree that the appropriate long term solution is to fix the DB modeling, but a fix that touches the DB is probably too high impact for this part of the release cycle.

Reply:
> We can't compare whether the router is the same, because the user could delete and recreate the same router relationship

@mansam @aufi that sounds like very rare, race condition kind of situation that we can fix with the more elegant solution involving DB Schema change.  Would the current simpler fix cover 99.99% of use cases?

Reply:
yeah, just comparing will probably get to 99% cases being correct. And the 1% will be fixed by next full refresh.

Reply:
we could also just set the test as pending, the current state is like 95% correct, where the 5% will be fixed by full refresh. Setting/changing router interfaces is a very rare operation, comparing to others, so it should not be an issue.